.github/workflows/prepare.yaml

```diff
@@ -0,0 +1,28 @@
+name: prepare
+
+on:
+  push:
+    branches: [master]
+    tags-ignore: ["v*"]
+  pull_request:
+    branches: [master]
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          submodules: recursive
+
+      - name: Install shellcheck
+        run: sudo apt install shellcheck
```

Comment:
Accepted, thanks. But what's the cause and reasoning for this?

Reply:
From `apt(8)`:

> SCRIPT USAGE AND DIFFERENCES FROM OTHER APT TOOLS
> The `apt(8)` commandline is designed as an end-user tool and it may change behavior between versions. While it tries not to break backward compatibility this is not guaranteed either if a change seems beneficial for interactive use.
> 
> All features of `apt(8)` are available in dedicated APT tools like `apt-get(8)` and `apt-cache(8)` as well.  `apt(8)` just changes the default value of some options (see `apt.conf(5)` and specifically the Binary scope). So you should prefer using these commands (potentially with some additional options enabled) in your scripts as they keep backward compatibility as much as possible.

While `DEBIAN_FRONTEND=noninteractive` is for `debconf` to not prompt for anything when configuring packages.